src/object/srv_obj.c

```diff
@@ -2920,6 +2920,22 @@ ds_obj_rw_handler(crt_rpc_t *rpc)
 	if (obj_rpc_is_fetch(rpc)) {
 		struct dtx_handle	*dth;
 
+		/* ORF_FETCH_EPOCH_EC_AGG_BOUNDARY only used for rebuild fetch. The container's
+		 * sc_ec_agg_eph_boundary possibly be different on the initiator and target engines
+		 * of the rebuild fetch, initiator selected fetch epoch possibly lower than readable
+		 * epoch at target engine side if vos aggregation merged adjacent extents to higher
+		 * epoch. For this case increase the fetch epoch to sc_ec_agg_eph_boundary.
+		 */
+		if (orw->orw_flags & ORF_FETCH_EPOCH_EC_AGG_BOUNDARY) {
+			if (orw->orw_epoch < ioc.ioc_coc->sc_ec_agg_eph_boundary) {
+				D_DEBUG(DB_IO, DF_UOID" increase fetch epoch from "DF_X64" to "
+					DF_X64" (sc_ec_agg_eph_boundary)\n",
+					DP_UOID(orw->orw_oid), orw->orw_epoch,
+					ioc.ioc_coc->sc_ec_agg_eph_boundary);
+				orw->orw_epoch = ioc.ioc_coc->sc_ec_agg_eph_boundary;
```

Comment:
What if EC agg boundary > rebuild epoch (spc_rebuild_fence)? EC aggregation doesn't merge partial updates across rebuild epoch, so we'd use the min(rebuild_epoch, ec_agg_boundary) here?

Reply:
hmm, right, let me refresh it, thanks.